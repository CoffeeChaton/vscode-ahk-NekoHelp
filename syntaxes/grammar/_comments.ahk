;https://www.autohotkey.com/docs/v1/Language.htm#comments
; This entire line is a comment.
Run Notepad ; This is a comment on the same line as a command.
Run Notepad; not comment

/*
MsgBox, This line is commented out (disabled).
MsgBox, Common mistake: */ this does not end the comment.
MsgBox, This line is commented out. 
*/

#NoEnv ;Comment
#Hotstring EndChars -()[]{}':;"/\,.?!`n `t ;Comment

#Hotstring EndChars - ()[]{}': ;Comment

MsgBox, % Abs(-1.2) ;Comment Returns 1.2
