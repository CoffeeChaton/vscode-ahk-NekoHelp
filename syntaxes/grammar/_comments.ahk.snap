>;https://www.autohotkey.com/docs/v1/Language.htm#comments
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>; This entire line is a comment.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>Run Notepad ; This is a comment on the same line as a command.
#^^^ source.ahk meta.command.line.ahk support.function.ahk
#   ^^^^^^^^ source.ahk meta.command.line.ahk
#           ^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk
#            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>Run Notepad; not comment
#^^^ source.ahk meta.command.line.ahk support.function.ahk
#   ^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.command.line.ahk
>
>/*
#^^ source.ahk comment.block.ahk
>MsgBox, This line is commented out (disabled).
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk comment.block.ahk
>MsgBox, Common mistake: */ this does not end the comment.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk comment.block.ahk
>MsgBox, This line is commented out. 
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk comment.block.ahk
>*/
#^^ source.ahk comment.block.ahk
>
>#NoEnv ;Comment
#^^^^^^ source.ahk string.parameter.directives.ahk keyword.control.directives.ahk
#      ^ source.ahk string.parameter.directives.ahk meta.comment.line.semicolon.ahk
#       ^^^^^^^^ source.ahk string.parameter.directives.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>#Hotstring EndChars -()[]{}':;"/\,.?!`n `t ;Comment
#^^^^^^^^^^ source.ahk string.parameter.directives.ahk keyword.control.directives.ahk
#          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.parameter.directives.ahk
#                                          ^ source.ahk string.parameter.directives.ahk meta.comment.line.semicolon.ahk
#                                           ^^^^^^^^ source.ahk string.parameter.directives.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>
>#Hotstring EndChars - ()[]{}': ;Comment
#^^^^^^^^^^ source.ahk string.parameter.directives.ahk keyword.control.directives.ahk
#          ^^^^^^^^^^^^^^^^^^^^ source.ahk string.parameter.directives.ahk
#                              ^ source.ahk string.parameter.directives.ahk meta.comment.line.semicolon.ahk
#                               ^^^^^^^^ source.ahk string.parameter.directives.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>
>MsgBox, % Abs(-1.2) ;Comment Returns 1.2
#^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#      ^ source.ahk meta.command.line.ahk punctuation.ahk
#       ^ source.ahk meta.command.line.ahk
#        ^ source.ahk meta.command.line.ahk punctuation.definition.variable.percent.ahk
#         ^ source.ahk meta.command.line.ahk
#          ^^^ source.ahk meta.command.line.ahk support.function.ahk
#             ^ source.ahk meta.command.line.ahk punctuation.bracket.ahk
#              ^ source.ahk meta.command.line.ahk keyword.operator.arithmetic.ahk
#               ^^^ source.ahk meta.command.line.ahk constant.numeric.decimal.ahk
#                  ^ source.ahk meta.command.line.ahk punctuation.bracket.ahk
#                   ^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk
#                    ^^^^^^^^^^^^^^^^^^^^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>